ultimate-pipeline/src/ultimate_pipeline/pipelines/data_processing/nodes.py

```diff
@@ -37,13 +41,32 @@ def convert_supervisely_annotations_to_yolo_format_dataframe(
         annotation_partitions (dict) - a dictionary (keyed by file names) containing content generator functions (Callables generating JSON annotation content as dict)
         meta_file (dict) - JSON content of meta.json
     """
-    logger.info("Reading folder data")
+    logger.info("Reading supervisely annotations folder data for detect data extraction")
+    annotation_partitions_without_extension = {}
+    for filename, content_generator in annotation_partitions.items():
+        filename_without_extension = filename[:filename.rfind(".")]
+        annotation_partitions_without_extension[filename_without_extension] = content_generator
+
+    yolo_detect_df = convert_images_annotations_folder_to_detect_data(source=annotation_partitions_without_extension, meta_file=meta_file)
+    columns_to_include = ["cls", "x", "y", "w", "h", "frame"]
```

Comment:
Moved column filtering here so that we can re-use a generic dataset factory in `catalog.yml` when persisting the dataset